frameworks/MLPlan/exec.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import math
+import os
+import json
+
+from frameworks.shared.callee import call_run, result, output_subdir, utils
+
+log = logging.getLogger(__name__)
+
+def run(dataset, config):
+    log.info("\n**** ML-Plans ****\n")
+
+    is_classification = config.type == 'classification'
+    
+    # Mapping of benchmark metrics to Weka metrics
+    metrics_mapping = dict(
+        acc='ERRORRATE',
+        auc='AUC',
+        logloss='LOGLOSS',
+        f1='F1',
+        r2='R2',
+        rmse='ROOT_MEAN_SQUARED_ERROR',
+        mse='MEAN_SQUARED_ERROR',
+        rmsle='ROOT_MEAN_SQUARED_LOGARITHM_ERROR',
+        mae='MEAN_ABSOLUTE_ERROR'
+    )
+    metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if metric is None:
+	    raise ValueError('Performance metric {} is not supported.'.format(config.metric))
+    
+    train_file = dataset.train.path
+    test_file = dataset.test.path
+    # Weka requires target as the last attribute
+    if dataset.target.index != len(dataset.predictors):
+        train_file = reorder_dataset(dataset.train.path, target_src=dataset.target.index)
+        test_file = reorder_dataset(dataset.test.path, target_src=dataset.target.index)
```

Comment:
you can now remove this here as the reordering has been done in the parent process
```suggestion
 ```